### ISSUE TITLE
Misleading error message about string literal occurs alongside real missing property error

### ISSUE THREAD

**Issue by matthieugicquel** [microsoft/TypeScript#37112]
**TypeScript Version:**  3.8 & Nightly in playground

**Search Terms:** TS2322 TS2741 String literals inference cannot infer type 'string'

**Code**
```ts
type Color = "white" | "black";
interface Surface {
    id: number;
    color: Color;
}

function useSurfaces(surface: Array<Surface>): void {};
useSurfaces([
  { id: 1, color: "white" }, // <-- Error TS2322 here, should be no error
  {color: "black"}, // <-- And here, should be TS2741
  ]);
```

**Expected behavior:**
A function call with a missing property should result in error TS2741 ("`Property 'id' is missing...`"), which explains the problem.

**Actual behavior:**
In the example above, error TS2741 isn't displayed, and instead there are multiple errors TS2322 ("`Type 'string' is not assignable to type...`"), even though this error is not justified in our case.

**Playground Link:** [Here](https://www.typescriptlang.org/play/?ts=nightly&ssl=8&ssc=47&pln=1&pc=1#code/C4TwDgpgBAwg9gGzgJygXigIgO4AsCWwEmUAPlgEYICGAxgNaYDcAUC-gHZHIBmd0AZQCuvflADeLKNKj4AJgC4oHIQFsKEZKxlRaiFEvhItLAL5sA9BagB1XBA5QAklGDIQruLuQRqRVwQAzlBwFABWELTAAHSxLHocgcBQgSJ8tBBKAILIyNQgADzCohkAfOhQANpSErKKUACMADS6+shKOAREJKZNNeKtxh1UdIxQvTUAuixWzlAA5hDJwPZYyPjzuMAkmsgGUAAqAgBMAOwALA1sM9YAQkLJeA6u4ND4wZw8uxByUDx7qig1D+Qg4UXwcEccggPE4hAhjmwhFwQMc1Fy+Vi0RYPFB4MhUCEgQgxXSEECAApUiVMlAcnlCqT+KUAJRKABucHkElMrCJJLS-Ep1WkA3kSmag32nUIxHGfVFeiGlBoDEwE2kkxZrFmLkWyV2KEOJwAzMdjq4kRkWtQOL8XHJIQByZL641nS4tPD4Wgo1TUejk2TJdmaDy4dG-YBeaEUITzNhAA)

**Related Issues:** Multiple issues about error TS2322 but none about the fact that it "replaces" TS2741 while it should not

**Comment by uhyo** [comment]
It seems that this bug was fixed in 3.9. I coundn't reproduce this bug in TS 3.9 or later.

**Comment by RyanCavanaugh** [comment]
Yep, seems fixed now. Thanks!
